### PR TITLE
[16.0][MIG] product_configurator: Migration to 16.0

### PR DESCRIPTION
**Pull request by dreispt**
Closes #96

### CODE REVIEW

**Comment by dreispt** [comment]
@dannyadair This has the build fixed, and should be mergeable.
Does it make sense to go ahead with the merge? The original OR has no approvals.

Also, I am skeptical these modules will work on v16, because I understand that views are now cached after being generated the first time. But I could be wrong.

File: product_configurator/wizard/product_configurator.py
Comment (SirAionTech) [inline product_configurator/wizard/product_configurator.py]: Just curious, why this change? This causes a lots of changes in the tests too

**Review by SirAionTech — CHANGES_REQUESTED** [review]
Thanks for the PR!
I'm just now beginning to work on this module and looks great, I tried it and seems to be working good.
I have reviewed the code and suggested some changes in https://github.com/ursais/product-configurator/pull/3, let me know what you think.

Could you please squash bot commits as explained in https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests? This PR already has a lot of commits.

I see there are also some edits due to https://github.com/OCA/product-configurator/pull/100, could you please remove them? A rebase on `16.0` might be enough since the PR has been merged.

For some reason the maintainer @PCatinean has not been tagged in the PR (doing it now), what do you think of this migration?

EDIT: I also noticed that the changes of https://github.com/OCA/product-configurator/pull/92 are missing, could you please include them?

File: product_configurator/wizard/product_configurator.py
Comment (dannyadair) [reply product_configurator/wizard/product_configurator.py]: I just took that from the original PR assuming there were Javascript changes that made that necessary

**Comment by dannyadair** [comment]
Thank you @dreispt this actually works well, also @SirAionTech has great improvements would be great if this could be merged in with his suggested adjustments

**Comment by dreispt** [comment]
@SirAionTech @dannyadair Proposed changes merged. If I get approval I can go ahead with the merge.

**Comment by dreispt** [comment]
@Murtaza-OSI

**Review by dannyadair — APPROVED** [review]
Thank you @dreispt LGTM

**Review by Murtaza-SerpentCS — APPROVED** [review]

**Pull request merged**
